arithmetization/src/main/java/net/consensys/linea/zktracer/container/stacked/list/StackedList.java

```diff
@@ -32,14 +35,100 @@
  *
  * @param <E> the type of elements stored in the list
  */
-public class StackedList<E> implements List<E>, StackedContainer {
-  private final Stack<List<E>> lists = new Stack<>();
+public class StackedList<E extends ModuleOperation> implements List<E>, StackedContainer {
+  private class CountedList extends ArrayList<E> {
```

Comment:
I understand that `CountedList` shouldn't be used outside of this class, but still I would put it in a separate file with `package-private` visibility, because it is not a small piece of code and would be cleaner.

Reply:
Fair point